src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java

```diff
@@ -133,14 +133,24 @@ public boolean exists(Path f) throws IOException {
    */
   @Override
   public boolean isDirectory(Path f) throws IOException {
-    LOG.debug("is directory: {}", f.toString());
-    return false;
+    try {
+      FileStatus fileStatus = getFileStatus(f);
```

Comment:
@sadekjb Can you please add "if not temporary path then ...". This way we will make sure that all this "isDirectory", "isFile" will not be activated on the path that contains temporary pattern. You can see other methods, how to make this if statement. thanks